src/bsoncxx/json.cpp

```diff
@@ -274,7 +274,12 @@ std::string to_json(types::value value) {
 
     json_visitor v(ss, false, 0);
 
-    switch ((int)value.type()) {
+    if (!value.type()) {
+        // TODO: use bsoncxx error category
+        throw std::runtime_error("invalid value");
```

Comment:
A user could default-construct a types::value object, or move from one, then try to pass it to to_json.